### PR TITLE
Add Linter

### PR DESCRIPTION
**Pull request by alisa911**

### CODE REVIEW

**Comment by alisa911** [comment]
https://github.com/osmandapp/OsmAnd-Issues/issues/2101

File: map/src/context/AppContext.js [hunk 1]
```diff
@@ -80,42 +80,37 @@ export const toHHMMSS = function (time) {
...
         return f.details.analysis.startTime;
     }
-    let dt = f.name.match(/(20\d\d)-(\d\d)-(\d\d)/)
+    let dt = f.name.match(/(20\d\d)-(\d\d)-(\d\d)/);
     if (!dt) {
-        dt = f.name.match(/(20\d\d)(\d\d)(\d\d)/)
+        dt = f.name.match(/(20\d\d)(\d\d)(\d\d)/);
     }
-    try {
+    if (dt) {
```
Comment (RZR-UA) [inline map/src/context/AppContext.js:93]: Please check with original code author and possibly get back try {}

File: map/src/drawer/components/favorite/FavoriteGroup.jsx [hunk 2]
```diff
@@ -137,47 +130,53 @@ export default function FavoriteGroup({index, group, enableGroups, setEnableGrou
...
+                        ctx.favorites[group.name]?.addToMap === true && enableGroups.some((e) => e.name === group.name)
+                    }
+                    onChange={(e) => {
+                        enableLayerWithGroups(setLoadingFavorites, e.target.checked).then();
+                    }}
+                    onClick={(e) => {
+                        e.stopPropagation();
+                    }}
+                />
+                {markers === [] ? <></> : favoritesPointsOpen ? <ExpandLess /> : <ExpandMore />}
```
Comment (RZR-UA) [inline map/src/drawer/components/favorite/FavoriteGroup.jsx]: attention

File: map/src/util/HttpApi.js [hunk 3]
```diff
@@ -229,16 +235,18 @@ export async function apiPost(url, data = '', options = null) {
...
+        // FormData, keep type=null
         body = data; // type is formed by fetch()
-    } else { // finally, try to convert from json
+    } else {
+        // finally, try to convert from json
         if (typeof data === 'object') {
             let converted = '';
-            try { converted = JSON.stringify(data); } catch { }
-
+            converted = JSON.stringify(data);
```
Comment (RZR-UA) [inline map/src/util/HttpApi.js:249]: get back try {}

**Pull request merged**